### ISSUE TITLE
Diagnostics not showing up for Clojure file

### ISSUE THREAD

**Issue by borkdude** [gtache/intellij-lsp#118]
I have written a linter for Clojure that supports LSP.

Configuration:
The server can be started with: `java -jar /tmp/clj-kondo.lsp-standalone.jar`.
I have configured this as a "raw command" to be triggered on the extensions `clj;cljs;cljc;edn`.

Attachment:  [clj-kondo.lsp-standalone.jar.zip](https://github.com/gtache/intellij-lsp/files/3835411/clj-kondo.lsp-standalone.jar.zip)

When I open a Clojure file and write this:

`(inc "foo")` an error should pop up saying: `Expected: number, received: string.`

When I enable logging, there is indeed a message indicating this error:

``` json
{
  "jsonrpc": "2.0",
  "method": "textDocument/publishDiagnostics",
  "params": {
    "uri": "file:///Users/borkdude/Dropbox/dev/clojure/clj-kondo/src/clj_kondo/core.clj",
    "diagnostics": [
      {
        "range": {
          "start": {
            "line": 12,
            "character": 5
          },
          "end": {
            "line": 12,
            "character": 5
          }
        },
        "severity": 2,
        "source": "clj-kondo",
        "message": "Expected: number, received: string."
      }
    ]
  }
}
```

But no error is showing up in the file itself:

<img width="189" alt="Screenshot 2019-11-12 10 45 37" src="https://user-images.githubusercontent.com/284934/68660497-920fb880-0539-11ea-8d0c-c9b8ddb5fe2f.png">

The source of the language server: 

https://github.com/borkdude/clj-kondo.lsp/tree/master/server

Note that the same language server works fine with a VSCode extension:

https://marketplace.visualstudio.com/items?itemName=borkdude.clj-kondo

Playing with the locations a bit. When I increment the column number, something appears, but it seems to be one column too far to the right. Decrementing the column number again doesn't show anything.

<img width="371" alt="Screenshot 2019-11-12 11 17 50" src="https://user-images.githubusercontent.com/284934/68662931-12d0b380-053e-11ea-8839-62366b994993.png">

**Comment by jeffmaury** [comment]
Not sure this is related as your diagnostic has an empty range but if you say it's is working in VSCode with the same LS, then it's worth trying my PR

**Comment by borkdude** [comment]
From the VSCode LSP Guide:

> Diagnostics Tips and Tricks
> If the start and end positions are the same, VS Code will underline with a squiggle the word at that position.
[source](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#diagnostics-tips-and-tricks)

**Comment by borkdude** [comment]
@jeffmaury I'd be happy to try out your PR but I have no idea how to build this project. Where can I read about this? I have successfully compiled the `intellij-lsp` subproject in IntelliJ now.

**Comment by gtache** [comment]
Indeed, the diagnostic range length is 0 so nothing is shown. The support for 0-length diagnostics is however implemented in 1.6.0, so you could try building this version.

**Comment by borkdude** [comment]
@gtache I'd like to give that a try, but I'm missing build instructions. What should I do to compile your project?

**Comment by borkdude** [comment]
@gtache Confirming that this worked using the released 1.6.0 zip file! I could not find it in the marketplace yet, so I installed it using a manual download.

https://github.com/borkdude/clj-kondo/blob/master/doc/editor-integration.md#lsp-server

**Comment by gtache** [comment]
Great! Yeah it hasn't been accepted yet, it always takes around 2 days for it to be available.
